api/src/api/agencies_v1/agency_schema.py

```diff
@@ -0,0 +1,77 @@
+from src.api.schemas.extension import Schema, fields
+from src.api.schemas.response_schema import AbstractResponseSchema, PaginationMixinSchema
+from src.pagination.pagination_schema import generate_pagination_schema
+
+
+class AgencyFilterV1Schema(Schema):
+    agency_id = fields.Integer()
+
+
+class AgencyListRequestSchema(Schema):
+    filters = fields.Nested(AgencyFilterV1Schema())
+    pagination = fields.Nested(
+        generate_pagination_schema(
+            "AgencyPaginationV1Schema",
+            ["created_at"],
+        ),
+        required=True,
+    )
+
+
+class AgencyContactInfoSchema(Schema):
+    """Schema for agency contact information"""
+
+    contact_name = fields.String()
+    address_line_1 = fields.String()
+    address_line_2 = fields.String(allow_none=True)
+    city = fields.String()
+    state = fields.String()
+    zip_code = fields.String()
+    phone_number = fields.String()
+    primary_email = fields.String()
+    secondary_email = fields.String(allow_none=True)
+
+
+class AgencyResponseSchema(Schema):
+    """Schema for agency response"""
+
+    agency_id = fields.Integer()
+    agency_name = fields.String()
+    agency_code = fields.String()
+    sub_agency_code = fields.String(allow_none=True)
+    assistance_listing_number = fields.String()
+    agency_submission_notification_setting = fields.String()  # Enum value
+
+    # Agency contact info as nested object
+    agency_contact_info = fields.Nested(AgencyContactInfoSchema, allow_none=True)
+
+    # Boolean flags
+    is_test_agency = fields.Boolean()
+    is_multilevel_agency = fields.Boolean()
+    is_multiproject = fields.Boolean()
+    has_system_to_system_certificate = fields.Boolean()
+    can_view_packages_in_grace_period = fields.Boolean()
+    is_image_workspace_enabled = fields.Boolean()
+    is_validation_workspace_enabled = fields.Boolean()
+
+    # Optional fields
+    ldap_group = fields.String(allow_none=True)
+    description = fields.String(allow_none=True)
+    label = fields.String(allow_none=True)
+
+    # Related agency info
+    top_level_agency_id = fields.Integer(allow_none=True)
```

Comment:
We can exclude the ID, although we will need this API to return the top level agency info for the frontend to build a hierarchy.

Reply:
I removed this but sounds like it should stay, though?

Reply:
We don't need the ID since that is purely internal, but the frontend would want the top level agency itself - which might be a ticket itself as that requires nesting the schema in itself.

You should be able to do that https://marshmallow.readthedocs.io/en/stable/nesting.html#nesting-a-schema-within-itself

Something like:
```
class AgencySchema(Schema):
     ...
    top_level_agency = fields.Nested(lambda: AgencySchema(exclude=("top_level_agency",)))
```
That'll nest it within itself, but only go one layer deep since you will have excluded nesting in the layer down.